wasp-ai/src/client/pages/ResultPage.jsx

```diff
@@ -393,6 +406,87 @@ function getCardinalNumber(number) {
   }
 }
 
+export function StarOurRepoModal({ isStarRepoOpen, setIsStarRepoOpen, appGenerationResult }) {
+  const [tokens, setTokens] = useState(0)
+  const { data: numTotalProjects } = useQuery(getNumProjects, {}, { enabled: isStarRepoOpen })
+
+  const tokenNumberStr = appGenerationResult?.project?.logs?.filter(log => log.content.toLowerCase().includes("tokens usage") === true)[0]?.content.split(':')[1]?.trim()
```

Comment:
Not bad! I would maybe prefer regex, since it is often more robust, but this can also work.

Reply:
WHy is this not in "useEffect"? If I am correct this will execute on every component render, while you use it only once, on first component render (since it is in useEffect)?